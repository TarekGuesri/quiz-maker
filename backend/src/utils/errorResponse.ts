export class ErrorResponse extends Error {
  statusCode = 0;
  constructor(message: string, statusCode: number) {
    super(message);
    this.statusCode = statusCode;

    Error.captureStackTrace(this, this.constructor);
  }
}
