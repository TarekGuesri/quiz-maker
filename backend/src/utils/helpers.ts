export const trimmer = (text: string): string => {
  if (text) {
    return text.trim();
  } else {
    return text;
  }
};
