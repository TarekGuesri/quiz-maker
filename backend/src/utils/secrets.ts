import dotenv from "dotenv";
import fs from "fs";

if (fs.existsSync(".env")) {
  console.log("Using .env file to supply config environment variables");
  dotenv.config({ path: ".env" });
} else {
  console.log("Using .env file to supply config environment variables");
  dotenv.config({ path: ".env.example" }); // you can delete this after you create your own .env file!
}

export const ENVIRONMENT = process.env.NODE_ENV;
const prod = ENVIRONMENT === "production"; // Anything else is treated as 'dev'

export const MONGODB_URI = prod
  ? process.env["MONGODB_URI"]
  : process.env["MONGODB_URI_LOCAL"];

if (!MONGODB_URI) {
  if (prod) {
    console.error(
      "No mongo connection string. Set MONGODB_URI environment variable.",
    );
  } else {
    console.error(
      "No mongo connection string. Set MONGODB_URI_LOCAL environment variable.",
    );
  }
  process.exit(1);
}
