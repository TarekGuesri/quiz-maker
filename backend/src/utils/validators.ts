import { Schema } from "express-validator";

export const quizSchema: Schema = {
  title: {
    // isLength: { errorMessage: "Title can't be empty", options: { min: 1 } },
    errorMessage: "Title can't be empty!",
    notEmpty: true,
  },
  questions: {
    isArray: { options: { min: 1, max: 10 } },
    errorMessage: "Questions must be an array of a length between 1 and 10!",
  },
  "questions.*.content": {
    errorMessage: "Question content can't be empty!",
    notEmpty: true,
  },
};
