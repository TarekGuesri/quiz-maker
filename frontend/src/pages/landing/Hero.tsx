import React from 'react';
import { makeStyles } from '@mui/styles';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';

const useStyles = makeStyles({
  root: {
    textAlign: 'center',
  },
});

interface HeroProps {}

export const Hero: React.FC<HeroProps> = ({}) => {
  const classes = useStyles();

  return (
    <Card sx={{ minWidth: 275 }} className={classes.root}>
      <CardContent>
        <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
          Word of the Day
        </Typography>
        <Typography variant="h5" component="div">
          benevolent
        </Typography>
        <Typography sx={{ mb: 1.5 }} color="text.secondary">
          adjective
        </Typography>
        <Typography variant="body2">
          well meaning and kindly.
          <br />
          {'"a benevolent smile"'}
        </Typography>
      </CardContent>
      <CardActions>
        <Button size="small">Learn More</Button>
      </CardActions>
    </Card>
  );
};
