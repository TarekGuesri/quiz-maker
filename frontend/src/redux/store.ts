import { Action, ThunkAction, configureStore } from "@reduxjs/toolkit";

import ui from "./ui/ui-slice";

export const store = configureStore({
  reducer: {
    ui,
  },
  devTools: true,
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
