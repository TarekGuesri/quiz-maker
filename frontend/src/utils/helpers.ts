export const timeFormatter = (
  seconds: number,
  minutes: number,
  hours: number,
): string => {
  const secondsString = seconds.toString().padStart(2, "0");
  const minutesString = minutes.toString().padStart(2, "0");

  if (hours === 0) {
    return `${minutesString}:${secondsString}`;
  } else {
    const hoursString = hours.toString().padStart(2, "0");

    return `${hoursString}:${minutesString}:${secondsString}`;
  }
};
