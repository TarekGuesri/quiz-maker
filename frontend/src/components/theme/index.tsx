import React, { FC } from 'react';
import CssBaseline from '@mui/material/CssBaseline';
import PropTypes from 'prop-types';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import { useAppSelector } from 'src/redux/hooks';
import varelaRound from 'src/assets/fonts/VarelaRound-Regular.ttf';

export const Theme: FC = (props) => {
  const { darkMode } = useAppSelector((state) => state.ui);

  const theme = createTheme({
    palette: {
      mode: darkMode ? 'dark' : 'light',
      background: {
        default: darkMode ? '#333' : '#f7f7f7',
        paper: darkMode ? '#333' : '#fbfbfb',
      },
      primary: {
        main: '#8e49de',
        // contrastText:
      },
      secondary: {
        main: '#f9de4a',
      },
      contrastThreshold: 3,
      tonalOffset: 0.2,
    },
    typography: {
      fontFamily: 'Varela Round',
    },
    components: {
      MuiCssBaseline: {
        styleOverrides: `
          @font-face {
            font-family: 'Varela Round';
            font-style: normal;
            font-display: swap;
            font-weight: 400;
            src: url(${varelaRound}) format('truetype');
        `,
      },
    },
  });

  return (
    <ThemeProvider theme={theme}>
      {props.children}
      <CssBaseline />
    </ThemeProvider>
  );
};

Theme.propTypes = {
  children: PropTypes.node.isRequired,
};
