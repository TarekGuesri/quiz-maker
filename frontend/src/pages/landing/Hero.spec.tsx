/**
 * @jest-environment jsdom
 */

import { cleanup, render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { Provider } from "react-redux";
import { Theme } from "src/components/theme";
import { store } from "src/redux/store";
import { history } from "src/utils/history";
import { Hero } from "./Hero";

describe("src/components/loading/Hero.tsx", () => {
  afterEach(() => cleanup());

  test("Render Loading component", () => {
    const renderedLoading = render(
      <Provider store={store}>
        <Router history={history}>
          <Theme>
            <Hero />
          </Theme>
        </Router>
      </Provider>,
    );

    expect(renderedLoading.container).toMatchSnapshot();
  });
});
