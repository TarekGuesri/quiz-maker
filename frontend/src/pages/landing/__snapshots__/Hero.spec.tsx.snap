// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/create-quiz/Hero.tsx Render Hero component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-root-6 css-4nwtfp-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root makeStyles-cardContent-7 css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h5 css-93lkom-MuiTypography-root"
      >
        Welcome to 
        <span
          class="makeStyles-brand-1"
        >
          QuizMaker
        </span>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1clpbiw-MuiTypography-root"
      >
        Here you can make your own quizzes and share them with others!
      </p>
      <div
        class="makeStyles-onlineTestBackground-5"
      >
        <img
          alt=""
          class="makeStyles-onlineTestImage-4"
          src="online-test.svg"
        />
      </div>
      <div
        class="makeStyles-actionBox-3 MuiBox-root css-0"
      >
        <a
          href="/create"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root makeStyles-button-2 css-17gny3x-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Create a Quiz
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
